util/process_file.py

```diff
@@ -0,0 +1,36 @@
+import datetime
+from typing import Callable
+from sqlalchemy import update
+from sqlalchemy.orm.session import Session
+
+from arlo_server.models import File  # type: ignore
+
+
+def process_file(session: Session, file: File, callback: Callable[[], None]) -> bool:
+    if file.processing_started_at:
+        return False
+
+    # Claim this file by updating the `processing_started_at` timestamp in such
+    # a way that it must not have been set before.
+    file.processing_started_at = datetime.datetime.utcnow()
+    result = session.execute(
```

Comment:
Does `session.execute` work before you call `session.commit`? Is that why this is different from how you would normally update the db through a model?

Reply:
the `commit` just commits the transaction, the SQL statements are still issued in the interim, and the update locking of the DB ensures that only one of these updates comes through. Great way to do this and be somewhat prepared for multi-threaded job processing!

Reply:
So if there were another thread/process/whatever that tried to grab the same one at the same time, they'd have different sessions/transactions and could both grab the same one, right? So maybe I do want an immediate commit?

Reply:
No, what would happen is that only one of the updates would go through, as any reasonable database (postgres included), would serialize the updates to the same row and only one thread would successfully get a rowcount return of 1.

There will be a problem if the thread processing the job fails and we need to retry, but that's a separate issue.